src/sqlfluff/dialects/dialect_hive.py

```diff
@@ -617,6 +617,36 @@ class TruncateStatementSegment(BaseSegment):
     )
 
 
+class SetStatementSegment(BaseSegment):
+    """A `SET` statement.
+
+    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+Commands
+    """
+    type = "set_statement"
+
+    match_grammar = Sequence(
+        "SET",
+        OneOf(
+            # set -v
+            Sequence(
+                StringParser("-", SymbolSegment, type="option_indicator"),
```

Comment:
Is it not relatively trivial to push this up to the lexer?

Reply:
Ah, actually, I can see how that'd clash - retracted

Reply:
Could do - it seemed overkill for such a small segment. There's an awkward overlap with numeric literals etc...